x-pack/plugins/security_solution/public/management/pages/policy/view/ingest_manager_integration/endpoint_policy_edit_extension.test.tsx

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { waitFor } from '@testing-library/react';
+import {
+  PackagePolicy,
+  NewPackagePolicy,
+  PACKAGE_POLICY_API_ROOT,
+  AGENT_API_ROUTES,
+} from '@kbn/fleet-plugin/common';
+
+import { EndpointDocGenerator } from '../../../../../../common/endpoint/generate_data';
+import { useEndpointPrivileges as _useEndpointPrivileges } from '../../../../../common/components/user_privileges/endpoint/use_endpoint_privileges';
+import { getEndpointPrivilegesInitialStateMock } from '../../../../../common/components/user_privileges/endpoint/mocks';
+import { AppContextTestRender } from '../../../../../common/mock/endpoint';
+import { policyListApiPathHandlers } from '../../store/test_mock_utils';
+import { createFleetContextRendererMock } from './mocks';
+import { EndpointPolicyEditExtension } from './endpoint_policy_edit_extension';
+
+jest.mock('../../../../../common/components/user_privileges/endpoint/use_endpoint_privileges');
+const useEndpointPrivilegesMock = _useEndpointPrivileges as jest.Mock;
+
+describe('When displaying the EndpointPolicyEditExtension fleet UI extension', () => {
+  let render: () => ReturnType<AppContextTestRender['render']>;
+  const generator = new EndpointDocGenerator();
+  const artifactCards = [
+    'trustedApps-fleet-integration-card',
+    'eventFilters-fleet-integration-card',
+    'hostIsolationExceptions-fleet-integration-card',
+    'blocklists-fleet-integration-card',
+  ];
+
+  beforeEach(() => {
+    const policyPackagePolicy = generator.generatePolicyPackagePolicy();
+    const policyListApiHandlers = policyListApiPathHandlers();
+
+    const mockedTestContext = createFleetContextRendererMock();
+    mockedTestContext.coreStart.http.get.mockImplementation((...args) => {
+      const [path] = args;
+      if (typeof path === 'string') {
+        if (path === `${PACKAGE_POLICY_API_ROOT}/${policyPackagePolicy.id}`) {
+          return Promise.resolve({
+            item: policyPackagePolicy,
+            success: true,
+          });
+        }
+
+        if (path === `${AGENT_API_ROUTES.STATUS_PATTERN}`) {
+          return Promise.resolve({
+            results: { events: 0, total: 5, online: 3, error: 1, offline: 1 },
+            success: true,
+          });
+        }
+
+        if (policyListApiHandlers[path]) {
+          return Promise.resolve(policyListApiHandlers[path]());
+        }
+      }
+
+      return Promise.reject(new Error(`unknown API call (not MOCKED): ${path}`));
+    });
```

Comment:
Just FYI: there are some Fleet http mocks you could use as well that would avoid doing this type of setup - here:

https://github.com/elastic/kibana/blob/3730dd0779ed8efd74aee88f57422781ec1ac122/x-pack/plugins/security_solution/public/management/pages/mocks/fleet_mocks.ts

Reply:
sweet, thanks for pointing that out 👍